crates/burn-jit/src/kernel/conv/conv2d/gemm/homogeneous/base.rs

```diff
@@ -0,0 +1,450 @@
+        #[comptime] has_bias: bool,
+    ) -> Self::AccumulatorLoader {
+        Self::AccumulatorLoader::new(bias, n_offset, config.to_smm_config(), has_bias)
+    }
+
+    fn init_unloader(out: &mut Tensor<Line<EG>>, x_offset: u32, y_offset: u32) -> Self::Out {
+        Self::Out::new(out, x_offset, y_offset, 0)
+    }
+
+    fn init_accumulator(#[comptime] config: Self::Config) -> Self::Accumulator {
+        SMM::init_accumulator(config.to_smm_config())
+    }
+}
+
+impl<EG, ES, Acc, SMM> ConvolutionKernel<EG, EG> for ImplicitGemmConvolution<EG, ES, Acc, SMM>
+where
+    EG: Numeric,
+    ES: Numeric,
+    Acc: Numeric,
+    SMM: stage::Matmul<ES, EG, Acc>,
+{
+    type Config = config::Config<SMM::Config>;
+
+    fn check_config(config: Self::Config) {
+        SMM::check_config(config.to_smm_config());
+    }
+
+    fn check_availability<R: Runtime>(
+        client: &ComputeClient<R::Server, R::Channel>,
+    ) -> Result<(), &str> {
+        SMM::check_availability::<R>(client)
+    }
+
+    fn make_config(
+        problem: &ConvolutionProblem,
+        cube_dim: &CubeDim,
+        cube_count: &CubeCount,
+        advanced_config: &AdvancedConfig,
+    ) -> Self::Config {
+        let smm_config = SMM::make_config(
+            &problem.as_matmul_problem(),
+            cube_dim,
+            cube_count,
+            advanced_config,
+        );
+
+        config::Config::new(
+            smm_config,
+            problem.m as u32 % SMM::M != 0,
+            problem.n as u32 % SMM::N != 0,
+            problem.k as u32 % SMM::K != 0,
+            problem.lhs_layout,
+            problem.rhs_layout,
+            problem.lhs_line_size as u32,
+            problem.rhs_line_size as u32,
+            problem.out_line_size as u32,
+            (problem.out_shape_y as u32, problem.out_shape_x as u32),
+            problem.kernel_size,
+            &problem.options,
+            problem.has_bias,
+        )
+    }
+}
+
+impl<
+        EG: Numeric,
+        ES: Numeric,
+        Acc: Numeric,
+        SMM: stage::Matmul<ES, EG, Acc, LhsReader = LhsReader<ES>, RhsReader = RhsReader<ES>>,
+    > ConvolutionLaunch<EG, EG> for ImplicitGemmConvolution<EG, ES, Acc, SMM>
+{
+    unsafe fn launch_unchecked<R: Runtime>(
+        client: &ComputeClient<<R as Runtime>::Server, <R as Runtime>::Channel>,
+        cube_dim: CubeDim,
+        cube_count: CubeCount,
+        input: TensorArg<'_, R>,
+        weight: TensorArg<'_, R>,
+        bias: TensorArg<'_, R>,
+        out: TensorArg<'_, R>,
+        config: <Self as ConvolutionKernel<EG, EG>>::Config,
+    ) {
+        Self::check_config(config);
+
+        launch::launch_unchecked::<EG, ES, Acc, Self, SMM, R>(
+            client,
+            cube_count,
+            cube_dim,
+            input,
+            weight,
+            bias,
+            out,
+            config,
+            config.has_bias,
+        );
+    }
+}
+
+#[cube(launch_unchecked)]
+pub(crate) fn launch<
```

Comment:
I would rename that to `implicit_conv` instead. It improves the name displayed in `ncu` and with our profiler.